exercises/wordy/Example.cs

```diff
@@ -1,51 +1,49 @@
 using System;
-using System.Collections.Generic;
 using System.Text.RegularExpressions;
 
 public static class Wordy
 {
-    private static readonly Regex VariableAndOperatorGroupsRegex =
-        new Regex(string.Format(@"{0} {1} {0}\s*{1}*\s*{0}*", @"(-?\d+)", "(plus|minus|multiplied by|divided by)"));
-    private static readonly Dictionary<string, Func<int, int, int>> Operations = new Dictionary<string, Func<int, int, int>>
-        {
-            { "plus", (x, y) => x + y },
-            { "minus", (x, y) => x - y },
-            { "multiplied by", (x, y) => x * y },
-            { "divided by", (x, y) => x / y },
-        };
+    private static readonly Regex EquationRegex = new Regex(
+        @"^What is (?<left>-?\d+)(?<operations> (?<operand>plus|minus|multiplied by|divided by) (?<right>-?\d+))*\?$", RegexOptions.Compiled);
 
     public static int Answer(string question)
     {
-        var parsedProblem = ParseProblem(question);
-        var firstPass = Operations[parsedProblem.Operation1](parsedProblem.X, parsedProblem.Y);
-        if (parsedProblem.Operation2.Length == 0)
-            return firstPass;
-        return Operations[parsedProblem.Operation2](firstPass, parsedProblem.Z);
+        var match = Parse(question);
+        if (match is null)
```

Comment:
I believe `Match` always returns a value but you should check `if (!match.Success)` to determine if there was nothing to match:

> You can determine whether the regular expression pattern has been found in the input string by checking the value of the returned Match object's Success property.

https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.match?view=netframework-4.7.2

Reply:
You are right. I've removed the superfluous check.